ci/jest-reporter.js

```diff
@@ -1,47 +1,96 @@
+}
+
+
+
 class MyCustomReporter extends BaseReporter {
   constructor(globalConfig, options) {
     super(globalConfig, options)
     this._globalConfig = globalConfig
     this._options = options
     this.runId = process.env.RUN_ID
+    this.taskArns = []
+    this.LogFiles = []
   }
 
   onRunStart(results, options) {
-    const message = buildDiscordStart(results, this.runId)
-    const data = { embeds: message, username: 'jest-reporter' }
-    const out = child_process.execSync(
-      `curl -X POST \
-        -H "Content-Type: application/json" \
-        -d '${JSON.stringify(data)}' \
-        ${process.env.DISCORD_WEBHOOK_URL}`
-    )
-    console.log(out.toString())
+    main_task().then(() => {
+      console.log('Done !!!')
+      this.taskArns = taskArns
+      this.LogFiles = generateDiscordCloudwatchLogFile(taskArns)
+      this.testPassUrl = process.env.DISCORD_WEBHOOK_URL_TEST_RESULTS
+      this.testFailUrl = process.env.DISCORD_WEBHOOK_URL_TEST_FAILURES
+
+      const message = buildDiscordStart(results, this.runId, this.LogFiles)
+      const data = { embeds: message, username: 'jest-reporter' }
+
+      const retryDelayMs = 300000 // 300k ms = 5 mins
+      sendDiscordNotification(this.testPassUrl, data, retryDelayMs)
+    })
+      .catch((error) => {
+        console.error(error)
+        process.exit(1)
+      })
   }
 
   onRunComplete(contexts, results) {
-    const message = buildDiscordSummary(results, this.runId)
+    const message = buildDiscordSummary(results, this.runId, this.LogFiles)
     const data = { embeds: message, username: 'jest-reporter' }
-    const out = child_process.execSync(
+
+    if (results.numFailedTestSuites > 0) {  // Test Failures
+      const out2 = child_process.execSync(     /* In future need to fix why sendDiscordNotification() used here for the second time like in onRunStart does not work here */
+        `curl -X POST \
+          -H "Content-Type: application/json" \
+          -d '${JSON.stringify(data)}' \
+          ${this.testFailUrl}
+        `
+      )
+      console.log(out2.toString())
+    }
+    
+    const out = child_process.execSync(     /* In future need to fix why sendDiscordNotification() used here for the second time like in onRunStart does not work here */
       `curl -X POST \
         -H "Content-Type: application/json" \
         -d '${JSON.stringify(data)}' \
-        ${process.env.DISCORD_WEBHOOK_URL}
+        ${this.testPassUrl}
       `
     )
     console.log(out.toString())
   }
 }
 
-function buildDiscordStart(results, runId) {
+function buildDiscordStart(results, runId, logFileName) {
   let startedAt = results.startTime
   try {
     startedAt = (new Date(results.startTime)).toGMTString()
   } catch {
     // pass
   }
+
+  let logFile = logFileName
+  if (logFileName.length === 0) {
+    logFile = ["No LogFile found!!!"]
+  }
+
+  let commitHashNames = 'js-ceramic (85d6c9789d28)\nipfs-daemon (85d6c9789d28)' // Placeholder work-in-progress
```

Comment:
If this is functionality is WIP I don't think we should merge it in. Should instead be put in a separate branch to be worked on there